lib/charms/loki_k8s/v0/loki_push_api.py

```diff
@@ -437,20 +566,46 @@ def __init__(
         Args:
             charm: a `CharmBase` object that manages this `LokiConsumer` object. Typically this is
                 `self` in the instantiating class.
-            relation_ name: a string name of the relation between `charm` and the Loki
-                charmed service.
+            relation_name: the string name of the relation interface to look up.
+                If `charm` has exactly one relation with this interface, the relation's
+                name is returned. If none or multiple relations with the provided interface
+                are found, this method will raise either an exception of type
+                NoRelationWithInterfaceFoundError or MultipleRelationsWithInterfaceFoundError,
+                respectively.
             alert_rules_path: an optional path for the location of alert rules files.
                 Defaults to "src/loki_alert_rules" at the top level of the
                 charm repository.
 
         Returns:
             Nothing.
         """
+        if not relation_name:
+            try:
+                # Check if there is just one relation with the right interface
+                relation_name = _get_single_relation_by_interface(
+                    charm, RELATION_INTERFACE_NAME, RelationDirection.REQUIRED
+                )
+            except NoRelationWithInterfaceFoundError:
+                raise ModelError(
+                    f"No required relation with the '{RELATION_INTERFACE_NAME}' interface found; "
+                    f"did you add a relation with the '{RELATION_INTERFACE_NAME}' interface in the charm's "
+                    "metadata.yaml file?"
+                )
+            except MultipleRelationsWithInterfaceFoundError as e:
+                raise ModelError(
+                    f"Multiple required relations with the '{RELATION_INTERFACE_NAME}' interface found: "
+                    f"{e.relations}; you must specify which relation should be managed by this "
+                    f"{type(self)} by providing the `relation_name` argument."
+                )
+
+        if not alert_rules_path:
+            alert_rules_path = resolve_dir_against_main_path("loki_alert_rules")
```

Comment:
- I would have "loki_alert_rules" as the default instead of None.
- `resolve_dir_against_main_path` should probably be called regardless. See also https://github.com/canonical/prometheus-operator/pull/141